src/main/webapp/scripts/map.js

```diff
@@ -49,31 +69,76 @@ function initMap() {
   map.addListener("click", e => {
     placeMarkerAndPanTo(e.latLng, map);
   });
+  
   /*Add Hard-Coded Markers */
   addMarker(map);
 }
 
-/* Temporary Function to Hold Marker for Sky Garden */
 function addMarker(map) {
```

Comment:
I am looking through addMarker then waitForClick and have a suggestion for the structure that might be worth considering :D 

The current implementation manually create the marker and add it to the map one by one. Given many recommendations, this will become an iteration which for each iteration will iterate on recommendations in waitForClick(). This perhaps can be improved since  both "marker" and "recommendations[i]" , whom we match through iteration in waitForClick(), have  more or less the same data, except place description
Instead of iterating and matching it,  maybe we can have all the details in recommendations[n] list only. This will make it simpler for the database as well. So, each recommendation in recommendations[n] list will have place name, type, lat, lng, description.

addMarker(map)  will:
1 . iterate through recommendations (currently hard-coded and can be easily fetched from datastore later with the same schema)
2. for each recommendation (can create a helper function or not) : create a marker, add evenListener to the marker from the recommendation details and add the marker to the map
     - the color from the icon can be obtained from the recommendation type (using switch statement)
     - lat lng converted to string to display on the html
     - name, category, and description can be displayed at is

With the suggestion above, we don't have to iterate recommendations detail for every marker added and can proceed everything in one iteration.

Reply:
Yes, this is the general idea I was thinking of following , but thank you for your explanation! This is just the first stage in creating the recommendations, so I thought it better to use four markers and to work on displaying them (as this took a lot of debugging!), than to focus on creating a scalable algorithm, but I can change it now.